src/plugins/crypto/gpg.c

```diff
@@ -166,16 +197,39 @@ static int getGpgBinary (char ** gpgBin, KeySet * conf, Key * errorKey)
 	}
 
 
-	// last resort
-	const size_t defaultLen = strlen (ELEKTRA_CRYPTO_DEFAULT_GPG_BIN) + 1;
-	*gpgBin = elektraMalloc (defaultLen);
+	// last resort number one - check for gpg2 at /usr/bin/gpg2
+	// NOTE this might happen if the PATH variable is empty
+	*gpgBin = strdup (ELEKTRA_CRYPTO_DEFAULT_GPG2_BIN);
+	if (!(*gpgBin))
+	{
+		ELEKTRA_SET_ERROR (87, errorKey, "Memory allocation failed");
+		return -1;
+	}
+
+	if (isExecutable (*gpgBin, NULL) == 0)
+	{
+		return 1;
```

Comment:
I expected you checked for errors, not for success. Can you add a small comment there?